src/mc_rbdyn/Robot.cpp

```diff
@@ -430,6 +430,8 @@ Robot::Robot(NewRobotToken,
     }
   }
 
+  createModuleFrames();
```

Comment:
This will fail to load some frames if `loadFiles` is false and some module frames are parented so this should be:
- called after loading surfaces in the case where `loadFiles` is true
- called afer copying surfaces in `::copyLoadedData` (nevermind, just saw this is already done as-such)

Reply:
> called after loading surfaces in the case where loadFiles is true

I'm not sure I get the point here?

Reply:
We get there after:
```
if(loadFiles)
{
  loadRSDFFromDir(...);
}
````

This will also create the frames associated to the surfaces. Otherwise these frames do no exist yet (until the surfaces are copied from another robot, that's the point of `loadFiles`)

If we load the frames from the RobotModule and if one of this frame uses a surface's frame as a parent then the frame won't be loaded although it is defined correctly.

Reply:
Got it, done in https://github.com/jrl-umi3218/mc_rtc/commit/27508c95af9192905f9fc4031436057c8dd6a01a